com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/model/VersionRule.java

```diff
@@ -40,4 +25,16 @@ public boolean allows(DeploymentType deploymentType) {
     public static VersionRule value(String caseInsensitiveValue) {
         return VersionRule.valueOf(caseInsensitiveValue.toUpperCase());
     }
+
+    private static boolean isSameOrHigher(DeploymentType deploymentType) {

```

Comment:
The names of these methods are not OK. `isSameOrHigher` implies that there's some sort of >= comparison in the method when there isn't. `isUpgradable` is inconsistent with `isSameOrHigher` in terms of wording and `isAll` is not even grammatically correct.